lib/util/utils.ts

```diff
@@ -84,15 +84,41 @@ export type ParsedOutput<T> = {
 
 /**
  * Parses the connection string and returns an object of type T.
+ *
+ * Connection strings have the following syntax:
+ *
+ * ConnectionString ::= Part { ";" Part } [ ";" ] [ WhiteSpace ]
+ * Part             ::= [ PartLiteral [ "=" PartLiteral ] ]
+ * PartLiteral      ::= [ WhiteSpace ] Literal [ WhiteSpace ]
+ * Literal          ::= ? any sequence of characters except ; or = or WhiteSpace ?
+ * WhiteSpace       ::= ? all whitespace characters including \r and \n ?
+ *
  * @param {string} connectionString The connection string to be parsed.
  * @returns {ParsedOutput<T>} ParsedOutput<T>.
  */
 export function parseConnectionString<T>(connectionString: string): ParsedOutput<T> {
-  return connectionString.split(';').reduce((acc, part) => {
+  return connectionString.trim().split(';').reduce((acc, part) => {
+    part = part.trim();
+    if (part === '') {
+      // parts can be empty
+      return acc;
+    }
+
     const splitIndex = part.indexOf('=');
+    if (splitIndex === -1) {
+      throw new Error("Connection string malformed: each part of the connection string must have an `=` assignment.");
+    }
+
+    const key = part.substring(0, splitIndex).trim();
+    if (key === '') {
+      throw new Error('Connection string malformed: missing key for assignment');
+    }
+
+    const value = part.substring(splitIndex + 1).trim();
+
     return {
       ...acc,
-      [part.substring(0, splitIndex)]: part.substring(splitIndex + 1)
+      [key]: value
```

Comment:
It feels weird to keep creating a new json object for each entry in the array. Replacing the `reduce` with a `forEach` can make this simpler with the object created once outside the loop, and key/value being added to it inside the loop

Reply:
Agree, and I dislike using reduce in general - it's almost always more confusing than a for loop. I was trying to preserve the original shape of the code but I'm more than happy to replace this with a more typical loop.

Reply:
@ramya-rao-a now using for-of, and it's wonderful. Let me know what you think.